modules/ximgproc/src/edgepreserving_filter.cpp

```diff
@@ -0,0 +1,237 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level
+// directory of this distribution and at http://opencv.org/license.html.
+//
+//                    Created by Simon Reich
+//
+
+#include "precomp.hpp"
+
+namespace cv
+{
+namespace ximgproc
+{
+using namespace std;
+
+void edgepreservingFilter(const InputArray _src, OutputArray _dst, int d,
+                          double threshold)
+{
+	Mat src = _src.getMat();
+
+	// [re]create the output array so that it has the proper size and type.
+	_dst.create(src.size(), src.type());
+	Mat dst = _dst.getMat();
+	src.copyTo(dst);
+
+	CV_Assert(src.type() == CV_8UC3);
+
+	if (d < 3)
+		d = 3;
+	int subwindowX = d, subwindowY = d;
+
+	if (threshold < 0)
+		threshold = 0;
+
+	// number of image channels
+	int nChannel = src.channels();
+
+	vector<double> pixel(nChannel, 0);
+	vector<vector<double>> line1(src.rows, pixel);
+	vector<vector<vector<double>>> weight(src.cols,
+	                                      line1); // global weights
+	vector<vector<vector<double>>> imageResult(
+	    src.cols, line1); // global normalized image
+
+	// do algorithm
+	cv::Mat subwindow;
+	for (int posX = 0; posX < src.cols - subwindowX; posX++)
+	{
+		for (int posY = 0; posY < src.rows - subwindowY; posY++)
+		{
+			cv::Rect roi =
+			    cv::Rect(posX, posY, subwindowX, subwindowY);
+			subwindow = src(roi);
+			cv::Mat subwindow1 = src(roi);
+			cv::GaussianBlur(subwindow1, subwindow, cv::Size(5, 5),
```

Comment:
> subwindow = src(roi);

This line seems useless. Due `cv::GaussianBlur()` call behavior.

Reply:
Removed `subwindow = src(roi);`.